caikit_nlp/data_model/embedding_vectors.py

```diff
@@ -0,0 +1,176 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Data structures for embedding vector representations
+"""
+# Standard
+from dataclasses import dataclass, field
+from typing import List, Union
+import json
+
+# Third Party
+from google.protobuf import json_format
+import numpy as np
+
+# First Party
+from caikit.core import DataObjectBase, dataobject
+from caikit.core.exceptions import error_handler
+import alog
+
+log = alog.use_channel("DATAM")
+error = error_handler.get(log)
+
+
+@dataobject(package="caikit_data_model.caikit_nlp")
+@dataclass
+class PyFloatSequence(DataObjectBase):
+    values: List[float] = field(default_factory=list)
+
+
+@dataobject(package="caikit_data_model.caikit_nlp")
+@dataclass
+class NpFloat32Sequence(DataObjectBase):
+    values: List[np.float32]
+
+    @classmethod
+    def from_proto(cls, proto):
+        values = [np.float32(x) for x in proto.values]
+        return cls(values)
+
+
+@dataobject(package="caikit_data_model.caikit_nlp")
+@dataclass
+class NpFloat64Sequence(DataObjectBase):
+    values: List[np.float64]
+
+    @classmethod
+    def from_proto(cls, proto):
+        values = [np.float64(x) for x in proto.values]
+        return cls(values)
+
+
+@dataobject(package="caikit_data_model.caikit_nlp")
+@dataclass
+class Vector1D(DataObjectBase):
+    """Data representation for a 1 dimension vector of float-type data."""
+
+    data: Union[
+        PyFloatSequence,
+        NpFloat32Sequence,
+        NpFloat64Sequence,
+    ]
+
+    def __post_init__(self):
+        error.value_check(
+            "<NLP92989048E>",
+            hasattr(self.data, "values"),
+            ValueError("Vector1D requires a float sequence data object with values."),
+        )
+
+    @classmethod
+    def from_json(cls, json_str):
+        """JSON does not have different float types. Move data into data_pyfloatsequence"""
+
+        json_obj = json.loads(json_str) if isinstance(json_str, str) else json_str
+        data = json_obj.pop("data")
+        if data is not None:
+            json_obj["data_pyfloatsequence"] = data
```

Comment:
this seems a bit hacky. Looks like it would only assign value to `PyFloatSequence` as well right? 

@prashantgupta24 any recommendation?

Reply:
Yes this is very unfortunate.  I don't see to/from json working with the vector union without this, but 🤮

Reply:
also note:  it is only to pyfloat because coming from JSON it's just a number.  Not sure who would want a data object from JSON, but testing the to/from implementation requires a solution here.